src/gzip.rs

```diff
@@ -102,7 +102,7 @@ pub enum Method {
 }
 
 impl Method {
-    pub fn parse<'a>(i: &'a [u8]) -> parse::Result<'a, Self> {
+    pub fn parse(i: &[u8]) -> parse::Result<Self> {
```

Comment:
Oh interesting, I assumed an explicit <'a> was needed because of `parse::Result`'s definition!

Reply:
since there is only one input lifetime rustc can infer that the parse result's lifetime has to be the same :)